src/com/nkita/practice/collections/PersonStatisticHandler.java

```diff
@@ -176,27 +170,53 @@ public void checkFullStatisticWithoutDuplicates() {
         Set<Person> personSet = new LinkedHashSet<>(persons);
         System.out.println("Statistics with duplicate");
 
-        for (Person person : persons) {
-            personSet.add(person);
-            System.out.println(personSet);
-            return;
+        for (Person person : personSet) {
+            System.out.println(person);
         }
 
 
     }
 
 
-
     public void removeDuplicates() {
         Set<Person> personSet = new LinkedHashSet<>(persons);
         persons.clear();
         persons.addAll(personSet);
 
-        }
+    }
+
+    private List<Person> getPersonsListByAge(int age, Gender gender) {
+
+        List<Person> personsList = new ArrayList<>();
 
+        for (Person person : persons) {
+            if (person.getAge() == age && (gender == null || gender == person.getGender())) {
+                personsList.add(person);
+            }
+
+        }
+        return personsList;
 
     }
 
 
+    public void checkOldestPerson(Gender gender) {
 
+        int maxAge = 0;
+        for (Person person : persons) {
+            boolean ignoreGender = gender == null;
```

Comment:
Why this variable is in loop?
You create this variable on every loop iteration.